### PR TITLE
ThemesSiteSelectorModal: Fix siteSlug value of null

### LINKED ISSUE #9298: Themes: On first load in multi-site view, the ThemesSiteSelectorModal points to `null` instead of the selected site
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/design (from a fresh login session)
2. Choose a theme and click the ... menu
3. Click Try & Customize
4. Select a site in the dropdown
5. Click Try & Customize

#### What I expected
To open the customizer

#### What happened instead
screen shot 2016-11-11 at 7 51 01 am

Note that if you go back and try again the customizer loads fine...it's only on the initial site load that it's a problem.  cc: @ockham #9193 

#### Browser / OS version
OSX Chrome

#### Context / Source
#automated-testing

### PR DESCRIPTION
**Pull request by ockham**
Fixes #9298.

The fix consists of two parts:
* Add a `<QuerySites />` component to `<ThemesSiteSelectorModal />` so that it fetches a user's sites from the network and populates the corresponding Redux state.
* Have `<ThemesSiteSelectorModal />` store only the current option's name (instead of the entire option object) in component state. This way, when Redux state updates, the component's `options` prop will update and cause a re-render with updated `getUrl` and `action` fields. 

To test:
* Verify that #9298 is fixed. (Also make sure to reproduce that issue on `master`. Remember to log out and in again each time or you won't be able to repro!)

cc @hoverduck

### CODE REVIEW

**Comment by hoverduck** [comment]
LGTM!

File: client/my-sites/themes/themes-site-selector-modal.jsx [hunk 1]
```diff
@@ -39,8 +40,8 @@ const ThemesSiteSelectorModal = React.createClass( {
 	},
 
 	trackAndCallAction( site ) {
-		const action = this.state.selectedOption.action;
-		const theme = this.state.selectedTheme;
+		const { selectedOption: optionName, selectedTheme: theme } = this.state;
+		const action = this.props.options[ optionName ].action;
```
Comment (mcsf) [inline client/my-sites/themes/themes-site-selector-modal.jsx]: Super minor, but this could be `const { action } = …`

**Comment by mcsf** [comment]
As discussed synchronously, I was still able to reproduce 9298, notably by selecting a private site (and benefitting from a very slow API call to the sites endpoint):

9341-broken

**Comment by ockham** [comment]
> As discussed synchronously, I was still able to reproduce 9298, notably by selecting a private site (and benefitting from a very slow API call to the sites endpoint):

I would love to guard against that, but there doesn't seem to be an easy way to do so. The underlying `SiteSelectorModal` (where I would need to e.g. set a `disabled` attr for that CTA button) still derives all its data from `sites-list`, which obviously doesn't necessarily reflect the Redux state. Rewriting it to obtain its data from Redux isn't an option yet, either, since there's a lot of stuff for which there aren't any Redux counterparts yet.

I'm going to merge this since it's making things at least somewhat better -- in particular for users with not-that-many sites.

**Pull request merged**